src/components/custom/dashboard/client/pages/diet-tracker/CustomMealModal.jsx

```
@@ -0,0 +1,822 @@
+        {/* Enhanced Tab Navigation */}
+        <div className="flex gap-1 p-1 bg-gradient-to-r from-zinc-800/60 to-zinc-800/40 rounded-xl border border-zinc-700/50 shadow-lg">
+          {/* AI Scanner Tab - First and promoted */}
+          <button
+            onClick={() => setActiveTab("ai-scanner")}
+            className={`relative flex-1 flex items-center justify-center gap-2 px-4 py-4 rounded-lg text-sm font-semibold transition-all duration-300 ${
+              activeTab === "ai-scanner"
+                ? "bg-gradient-to-r from-[#FF6B00] to-[#FF8B20] text-white shadow-xl transform scale-105"
+                : "text-zinc-300 hover:text-white hover:bg-zinc-700/60"
+            }`}
+          >
+            <div className="relative">
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)